docs/examples/objc-api-walkthrough/objc-api-walkthrough/ViewController.m

```diff
@@ -0,0 +1,134 @@
+//
+//  ViewController.m
+//  objc-api-walkthrough
+//
+//  Created by James Nocentini on 27/06/2017.
+//  Copyright © 2017 couchbase. All rights reserved.
+//
+
+#import "ViewController.h"
+#include <CouchbaseLite/CouchbaseLite.h>
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    
+    // create database
+    NSError *error;
+    CBLDatabase* database = [[CBLDatabase alloc] initWithName:@"my-database" error:&error];
+    if (!database) {
+        NSLog(@"Cannot open the database: %@", error);
+    }
+    
+    // create document
+//    NSError* error;
+    CBLDocument* newTask = [[CBLDocument alloc] init];
+    [newTask setObject:@"task-list" forKey:@"type"];
+    [newTask setObject:@"todo" forKey:@"owner"];
+    [newTask setObject:[NSDate date] forKey:@"createAt"];
+    [database saveDocument: newTask error: &error];
+    
+    // mutate document
+    [newTask setObject:@"Apples" forKey:@"name"];
+    [database saveDocument:newTask error:&error];
+    
+    // typed accessors
+    [newTask setObject:[NSDate date] forKey:@"createdAt"];
+    NSDate* date = [newTask dateForKey:@"createdAt"];
+    
+    // database transaction
```

Comment:
We should say that it's a batch operation not transaction even though we also guarantee the transaction in CBL's batch operation.